メッセージ送信の非同期通信化

chamomile-jasmine opened the pull request:
# WHAT
 投稿機能の非同期機能を作成する

# WHY
メッセージを送信するたびに送信画面にリダイレクトしているため、ビューが毎回再描画されてしまう。非同期通信を使って、メッセージの送信を非同期で行えるようにする。

gyazo GIF
https://gyazo.com/142a9bcbaefb75d17f0f05562aa288f9

chamomile-jasmine committed:
first commit

chamomile-jasmine committed:
非同期通信の実装

差分のみ表示を追加するよう設定した。
（投稿機能の非同期機能を作成した）

chamomile-jasmine committed:
2nd差分のみ表示を追加するよう設定

・メッセージ送信の非同期通信化（メッセージを送信するたびに送信画面にリダイレクトしているため、ビューが毎回再描画されてしまうため）
・コメントアウトとデバッグ用の記載を消去した

TOMARUMARU commented on app/assets/javascripts/message.js:
```diff
@@ -0,0 +1,53 @@
+  // メッセージ表示のHTMLを生成
+  function buildHTML(message) {
+    var insertImage = '';
+    if (message.image.url) {
```
三項演算子を使いましょう

chamomile-jasmine replied:
三項演算子を使って記述しました。

TOMARUMARU commented on app/assets/javascripts/message.js:
```diff
@@ -0,0 +1,53 @@
+  $('#new_message').on('submit', function(e) {
+    e.preventDefault();
+    var formdata = new FormData(this);
+    console.log(formdata);
```
console.logは必ず消しましょう

chamomile-jasmine replied:
消去しました。

TOMARUMARU commented on app/assets/javascripts/message.js:
```diff
@@ -0,0 +1,53 @@
+    .done(function(data) {
+      var message = buildHTML(data);
+      $('.messages').append(message);
+      $('.form__message').val('');
```
resetメソッドを使いましょう

chamomile-jasmine replied:
resetメソッドを使いました。フォームの写真データがリセットされるようになりました。

miyagit replied:
から文字にするのは必要ないですね。

chamomile-jasmine replied:
$('.form__message').val('');は不要ということでしょうか？

chamomile-jasmine replied:
から文字にする記述を削除しました！

TOMARUMARU reviewed (commented):
public/uploads以下はignoreしましょう。
差分もこちらのPRから消しましょう

chamomile-jasmine committed:
非同期通信の修正

・public/uploads以下をignoreしました。また git rm -r public/uploads
を実行し、差分を削除しました。
・三項演算子を使って記述しました。
・console.logを消去しました。
・resetメソッドの記述を追加しました。

chamomile-jasmine committed:
非同期通信について軽微な修正

・から文字にする記述を削除しました

TOMARUMARU commented on app/assets/javascripts/message.js:
```diff
@@ -0,0 +1,51 @@
+    .done(function(data) {
+      var message = buildHTML(data);
+      $('.messages').append(message);
+      // $('.form__message').val('');
```
不要なコメントアウトは消しましょう

chamomile-jasmine replied:
失礼致しました。消去しました。

chamomile-jasmine committed:
非同期通信について軽微な修正2

・残っていた余計なコメントアウトを消去しました。

miyagit commented on public/.gitignore:
```diff
@@ -0,0 +1 @@
+uploads/
```
.gitignoreはpublicに入れるのでしょうか？

chamomile-jasmine replied:
もともとある.gitignoreに追記する形に修正して、publicに作成した.gitignoreは削除しました。

chamomile-jasmine committed:
.gitignoreの場所を修正

・publicに作成した.gitignoreを削除して、もともとあった.gitignoreに追記する形に修正しました。

miyagit commented on app/controllers/messages_controller.rb:
```diff
@@ -4,14 +4,23 @@ class MessagesController < ApplicationController
+        format.html{redirect_to group_messages_path(@group), notice: 'メッセージが送信されました'}
+        format.json
+      end
+      # redirect_to group_messages_path(@group), notice: 'メッセージが送信されました'
```
不必要なコメントアウトは削除しましょう。

chamomile-jasmine replied:
不必要なコメントアウトを削除しました。確認できる範囲で他のコメントアウトも削除しました。

chamomile-jasmine committed:
軽微な修正

・不必要なコメントアウトを削除しました。

miyagit reviewed (approved):
LGTM

chamomile-jasmine commented:
ありがとうございます！
